### PR TITLE
Complete implementation and unit tests

### PR DESCRIPTION
**Pull request by weixifan**
This should be the final PR for the implementation of the App-scoped features in the Project Management Service.
- Reworked the display name semantics to return `None` when it has never been set rather than the empty string.
- Add `__eq__` and `__hash__` methods to `AndroidAppMetadata` and `IosAppMetadata`; write the remaining unit tests.
- Remove `name` as a publicly visible property from `AndroidAppMetadata` and `IosAppMetadata`.

### CODE REVIEW

**Comment by hiranya911** [comment]
I'll wait for #219 to be merged and this branch to be sync'ed.

**Comment by weixifan** [comment]
#219 has been merged; please take a look.

**Comment by weixifan** [comment]
I originally wanted to have the `ANDROID_APP_METADATA` constant as a `@fixture` as with the last PR, but unfortunately I wanted to share them between the Android and iOS tests, so they may as well be a module-level constant at that point.

**Comment by weixifan** [comment]
The coverage is now at 100%.

File: tests/test_project_management.py [hunk 1]
```diff
@@ -1065,6 +1082,12 @@ def test_delete_sha_certificates_not_found(self, android_app):
         assert excinfo.value.detail is not None
         assert len(recorder) == 1
 
+    def test_raises_if_app_has_no_project_id(self):
```
Comment (hiranya911) [inline tests/test_project_management.py]: Can we use `testutils.run_without_project_id()` here as well?

Comment (weixifan) [reply tests/test_project_management.py]: Done.

**Pull request merged**